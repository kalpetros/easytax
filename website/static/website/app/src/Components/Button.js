import styles from '../../../css/src/Components/Button.css';

import React from 'react';

class Button extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };

    }
    componentDidMount() {
    }
    render() {
	return(
	    <button className={styles.button}>
              <i className="material-icons">{this.props.icon}</i>
	    </button>
	);
    }
}

export{Button};
