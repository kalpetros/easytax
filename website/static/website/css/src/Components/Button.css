.button {
    background-color: unset;
    display: inline-block;
    padding: 0.5rem;
    border: none;
    cursor: pointer;
}

.button:hover {
    background-color: #eee;
    border-radius: 50%;
    transition: background-color 0.5s linear;
}
