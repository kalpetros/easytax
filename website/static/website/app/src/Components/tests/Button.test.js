import {Button} from '../Button';

import renderer from 'react-test-renderer';

import React from 'react';

test('Renders a round button', () => {
  const component = renderer.create(
    <Button type="round"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a basic button', () => {
  const component = renderer.create(
    <Button type="basic"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a button without name or icon', () => {
  const component = renderer.create(
    <Button/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a button with a name', () => {
  const component = renderer.create(
    <Button name="Test"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a button with an icon', () => {
  const component = renderer.create(
    <Button icon="close"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a button with a name and an icon', () => {
  const component = renderer.create(
      <Button name="Test"
              icon="close"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
