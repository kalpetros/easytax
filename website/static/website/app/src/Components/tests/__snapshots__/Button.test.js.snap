// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a basic button 1`] = `
<button
  className="basic"
/>
`;

exports[`Renders a button with a name 1`] = `
<button
  className="basic"
>
  <span>
    Test
  </span>
</button>
`;

exports[`Renders a button with a name and an icon 1`] = `
<button
  className="basic"
>
  <i
    className="material-icons"
  >
    close
  </i>
  <span>
    Test
  </span>
</button>
`;

exports[`Renders a button with an icon 1`] = `
<button
  className="basic"
>
  <i
    className="material-icons"
  >
    close
  </i>
</button>
`;

exports[`Renders a button without name or icon 1`] = `
<button
  className="basic"
/>
`;

exports[`Renders a round button 1`] = `
<button
  className="round"
/>
`;
